src/diagnostics/core_diagnostics.jl

```diff
@@ -1349,3 +1349,95 @@ add_diagnostic_variable!(
         end
     end,
 )
+
+###
+# Convective Available Potential Energy (2d)
+###
+function compute_cape!(out, state, cache, time)
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    g = TD.Parameters.grav(thermo_params)
+
+    # Get surface parcel properties
+    surface_thermal_state = Fields.level(cache.precomputed.ᶜts, 1)
+    surface_q = TD.total_specific_humidity.(thermo_params, surface_thermal_state)
+    surface_θ_liq_ice = TD.liquid_ice_pottemp.(thermo_params, surface_thermal_state)
+
+    # Create parcel thermodynamic states at each level based on energy & moisture at surface
+    parcel_ts_moist = TD.PhaseEquil_pθq.(thermo_params, 
+                                        cache.precomputed.ᶜts.p, 
+                                        surface_θ_liq_ice, 
+                                        surface_q)
+    
+    # Calculate virtual temperatures for parcel & environment
+    parcel_Tv = TD.virtual_temperature.(thermo_params, parcel_ts_moist)
+    env_Tv = TD.virtual_temperature.(thermo_params, cache.precomputed.ᶜts)
+
+    # Calculate buoyancy from the difference in virtual temperatures
+    ᶜbuoyancy = cache.scratch.ᶜtemp_scalar
+    ᶜbuoyancy .= g .* (parcel_Tv .- env_Tv) ./ env_Tv
+
+    # Integrate positive buoyancy to get CAPE
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        Operators.column_integral_definite!(out, max.(ᶜbuoyancy, 0))
```

Comment:
Not sure if this (and below `min` for CIN) are valid. This should be integrated between surface and the level of free convection and between the level of free convection the equilibrium level. I think there might be some weird cases where you have more than one of those. See for example the discussion here: 

https://github.com/Unidata/MetPy/issues/862